### DISCUSSION TITLE
What's the point of --flavor?

### DISCUSSION THREAD

**Discussion by micheljung** [AngeloAvv/flutter_flavorizr#194] [General]
I'd understand:

    flutter run --flavor <flavorName> -t lib/main.dart

or:

    flutter run -t lib/main_<flavorName>.dart

but setting both:

    flutter run --flavor <flavorName> -t lib/main_<flavorName>.dart

Seems a bit pointless.

**Comment by AngeloAvv** [comment]
If you don't pass the --flavor argument, custom build execution at the native level won't be run.
If you don't pass the -t argument, you cannot set a variable that represents the correct flavour you're actually running, without having chance to customise the dart/flutter code accordingly

**Comment by AngeloAvv** [comment]
The flutter development team introduced the appFlavor accessor. Starting from flutter_flavorizr 2.3.0, there's no need to create multiple main.dart files, only the --flavor parameter is necessary to switch among flavors
